File: Android/shared/src/main/kotlin/shared/dto/objects/Notice.kt
@file:UseSerializers(
    UUIDSerializer::class,
    DateISOSerializer::class
)

package shared.dto.objects

import shared.serializers.DateISOSerializer
import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import shared.dto.enums.NoticeCategory
import shared.dto.protocols.Responsable
import shared.serializers.UUIDSerializer
import java.util.*

object Notice {
    data object Responses {
        @Serializable
        data class Full(
            val id: UUID,
            val titleKey: String,
            val messageKey: String,
            val parameters: String?,
            val category: NoticeCategory,
            val isRead: Boolean,
            val date: Date?,
        ) : Responsable
    }
}

File: Node/src/dto/objects/Notice.ts
import { ValidateNested } from "class-validator"
import { NoticeCategory } from "../enums/NoticeCategory"
import { UUID } from "../tsPrimitives/UUID"
import { IsoDate } from "../tsPrimitives/IsoDate"

export namespace Notice {
    
    export namespace Parameters {}

    export namespace Responses {        
        
        export class Full {
            @ValidateNested()
            id: UUID
            
            titleKey: string
            messageKey: string
            parameters?: string
            category: NoticeCategory
            isRead: boolean
            date?: IsoDate

            constructor(
                id: UUID,
                titleKey: string,
                messageKey: string,
                category: NoticeCategory,
                isRead: boolean,
                parameters?: string,
                date?: IsoDate,
            ) {
                this.id = id
                this.titleKey = titleKey
                this.messageKey = messageKey
                this.parameters = parameters
                this.category = category
                this.isRead = isRead
                this.date = date
            }
        }
    }
}